test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -163,4 +172,72 @@ describe('View existing record,', function() {
         });
     });
 
+    describe("For side panel table of contents in Record App", function() {
+
+        beforeAll(function() {
+            var url = browser.params.url + "/record/#" + browser.params.catalogId + "/" + testParams.sidePanelTest.schemaName + ":" + testParams.sidePanelTest.tableName +  "/id=" + testParams.sidePanelTest.id;
+            browser.get(url);
+            recSidePanelCat_5 = chaisePage.recordPage.getSidePanelItemById(5);
+            fiddlerBtn = element(by.className('sidePanFiddler')).element(by.tagName('i')),
+            chaisePage.waitForElement(fiddlerBtn);
+        });
+        it('Table of contents should be displayed by default', function(){
+            var recordSidePan = chaisePage.recordPage.getSidePanel();
+            expect(recordSidePan.isDisplayed()).toBeTruthy("Side Panel is not visible when page loads initially.");
+        });
+
+        it('On click of Related table name in TOC, page should move to the contents and open the table details', function(){
+            var rtTableHeading = chaisePage.recordPage.getRelatedTableHeading(testParams.sidePanelTest.tableToShow);
+
+            recSidePanelCat_5.click().then(function(className) {
+              // related table should be visible
+                expect(rtTableHeading.isDisplayed()).toBeTruthy("Category_5 heading is not visible.");
+                return rtTableHeading.getAttribute("class");
+            }).then (function(className) {
+                expect(className).toContain("panel-open", "Related table panel is not open when clicked through TOC.");
+            }).catch( function(err) {
+                console.log(err);
+                expect('Encountered an error').toBe('Please check the log', 'Inside catch block');
+            });
+        });
+
+        it('Record count along with heading should match for the panel and related table content should be in correct order', function(){
+            chaisePage.recordPage.getSidePanelTableTitles().then(function(tableNames){
+              for (var i=0; i<tableNames.length; i++){
+                tableNames[i] = tableNames[i].replace(/ +/g, " ");
+              }
+                expect(tableNames).toEqual(testParams.sidePanelTest.sidePanelTableOrder,"Order is not maintained for related tables in side panel");
+                expect(tableNames.length).toEqual(testParams.sidePanelTest.tocCount, "Count mismatch for number of related tables in side panel");
```

Comment:
This seems unnecessary since the previous one is comparing both arrays. You can first test the count and then whole array. This way if the length fails, it won't check the other `expect`. But that's up to you. Either remove this second expect or just move it up.